File: ruby/bob/bob.rb
class Bob
  def self.hey(text)
    if text =~ /[A-Z]{4,}/
      if text =~ /[?]\z/
        "Calm down, I know what I'm doing!"
      else
        'Whoa, chill out!'
      end
    elsif text =~ /[A-Z]{2,}/ && text =~ /[!]\z/
      'Whoa, chill out!'
    elsif text =~ /[?]\z/
      'Sure.'
    elsif text == ''
      'Fine. Be that way!'
    elsif text =~ /\s\s/
      if text =~ /[?]/
        'Sure.'
      elsif text =~ /[a-z]+/
        'Whatever.'
      else
        'Fine. Be that way!'
      end
    else
      'Whatever.'
    end
  end
end

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      input =~ ~r{[A-Z]!} -> "Whoa, chill out!"
      input =~ ~r{[A-Z]\?} -> "Calm down, I know what I'm doing!"
      String.ends_with?(input, "?") -> "Sure."
      input =~ ~r/[[:upper:]]{2,}/u -> "Whoa, chill out!"
      !(input =~ ~r{\w}) -> "Fine. Be that way!"
      true -> "Whatever."
    end
  end
end
